packages/core/crates/core-packet/src/path.rs

```diff
@@ -0,0 +1,82 @@
+use libp2p_identity::PeerId;
+use std::fmt::{Display, Formatter};
+
+#[cfg_attr(feature = "wasm", wasm_bindgen::prelude::wasm_bindgen)]
+#[derive(Debug, Clone)]
+pub struct Path {
+    hops: Vec<PeerId>,
+    valid: bool,
```

Comment:
In future, the path might be constructed in 3 ways 
- manual without validation (which assumes the path is already given as valid and no validation needs to be done)
- manual with validation (the path is constructed and the is validated)
-  automatic (graph algorithm already constructs a valid path)

Because path validation is a non-trivial operation which involves DB, the immutable type carries the information if the validation already took place.